compiler/rustc_parse/src/parser/ty.rs

```diff
@@ -328,8 +328,6 @@ impl<'a> Parser<'a> {
             // Reference
             self.expect_and()?;
             self.parse_borrowed_pointee()?
-        } else if self.eat_keyword_noexpect(kw::Typeof) {
```

Comment:
It may be nice to keep a precise error message here and recover into an error type

Reply:
done that I think? @fmease do you think this is relevant enough to do a crater run or sth like that here?